### DISCUSSION TITLE
View collection with one-to-many relation

### DISCUSSION THREAD

**Discussion by dimitarnikolovv** [pocketbase/pocketbase#3741] [Q&A] [answered]
I have "posts" and "users" collections where each post has a relation with a user.
I want to create a view collection in which I want to store each user's posts.

So far I have come up with this:

```sql
SELECT 
  u.id AS id,
  JSON_GROUP_ARRAY(p.id) as userPosts
FROM 
  users u, posts p
WHERE
  p.author = u.id
GROUP BY u.id
```

But I wonder if it is possible to have each post as a relation aka `userPosts` to be a one-to-many relation.

**Comment by ganigeorgiev** [comment, marked as answer]
> But I wonder if it is possible to have each post as a relation aka userPosts to be a one-to-many relation.

Custom view schema fields mapping is in my todo, but at the moment this is not supported.

**Reply by dimitarnikolovv** [reply]
In this case I wonder which would be more performant, using `getFullList()` with a filter, or `getOne()` 15 times. Let's say that I have 500 posts but only 15 are specific to the user.

**Reply by ganigeorgiev** [reply]
I'm not sure that I understand the context and what is the desired behavior.

`getFullList` in general is expected to be faster compared to calling 15+ times `getOne` mostly due to the network latency.

If you just want to retrieve the posts for a specific user you can pass a `filter` with the user id:

```js
const posts = await pb.collection("posts").getFullList({
    filter: `author="${yourUserId}"`
    
    // or you can bind placeholder parameters from a map (JS SDK v0.19.0+)
    // (https://github.com/pocketbase/js-sdk#binding-filter-parameters)
    filter: pb.filter("author={:id}", { id: yourUserId })
})
```
